util/reggen/field.py

```diff
@@ -425,22 +459,42 @@ def apply_alias(self, alias_field: 'Field', where: str) -> None:
         self.alias_target = alias_field.alias_target
 
     def scrub_alias(self, where: str) -> None:
-        '''Replaces sensitive fields in field with generic names
+        """Replaces sensitive fields in field with generic names
 
         This function can be used to create the generic field descriptions
         from full alias hjson definitions.
-        '''
+        """
         # These attributes are scrubbed. Note that the name is scrubbed in
         # register.py already.
-        self.desc = ''
+        self.desc = ""
         self.enum = []
         self.resval = 0
         self.tags = []
         self.alias_target = None
 
     def to_systemrdl(self, importer: RDLImporter) -> systemrdl.component.Field:
         rdl_t = importer.create_field_definition(self.name)
-        return importer.instantiate_field(rdl_t,
-                                          self.name,
-                                          self.bits.lsb,
-                                          self.bits.width())
+        field = importer.instantiate_field(
+            rdl_t, self.name.upper(), self.bits.lsb, self.bits.width()
+        )
+
+        swaccess = self.swaccess.to_systemrdl()
+        importer.assign_property(field, "sw", swaccess["sw"])
+        if "onread" in swaccess:
+            importer.assign_property(field, "onread", swaccess["onread"])
+        if "onwrite" in swaccess:
+            importer.assign_property(field, "onwrite", swaccess["onwrite"])
+
+        hwaccess = self.hwaccess.to_systemrdl()
+        importer.assign_property(field, "hw", hwaccess["hw"])
+
+        if self.resval is not None:
```

Comment:
I suppose at some point, we will want to define mubi enums in a common RDL file, so we can use the native property (`encode`? My memory is fuzzy) to describe the CSR's type. (for a future PR, surely)

Reply:
Yes, enums will come in a future PR